File: wonsang-yoon/2023-05-02-valid-anagram.js
var isAnagram = function(s, t) {
    //boundary check for string only and lowercase
    if(t.length > s.length || typeof t === 'string' || typeof s === 'string' 
    || !/^[a-z]+$/.test(s) || !/^[a-z]+$/.test(t))
        return false;
    //make an duplicate of s
    let temp = s;
    //iternate 't' and switch off duplicating character with empty string
    for(let i in t){
        //replace 
        temp = temp.replace(t[i],'');
    }
    //if 'temp' results contains nothing means it is a anagram
    return temp == ''? true: false
};

File: Ellen Jung/2023-05-02-valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) < 1 or len(t) > 5 * 10 ** 4:
            raise ValueError

        for x in s:
            if x in t:
                t.replace(x, '')
            else:
                return False
        return True
